### ISSUE TITLE
New versions of SAM should clean up old docker images

### ISSUE THREAD

**Issue by abrooksv** [aws/aws-sam-cli#2250]
### Description
Now that a rapid layer is built and version specific, old versions should be cleaned up.

### Steps to reproduce
Upgrade and run a local invoke

### Observed result

```
amazon/aws-sam-cli-emulation-image-dotnetcore2.1                                                    rapid-1.0.0                                              9e9ef66ff35e        2 months ago        989MB
amazon/aws-sam-cli-emulation-image-java8                                                            rapid-1.0.0                                              05211eee97a6        2 months ago        662MB
amazon/aws-sam-cli-emulation-image-java8                                                            rapid-1.1.0                                              05211eee97a6        2 months ago        662MB
amazon/aws-sam-cli-emulation-image-java11                                                           rapid-1.0.0                                              8de94beb46f4        2 months ago        417MB
amazon/aws-sam-cli-emulation-image-java11                                                           rapid-1.1.0                                              8de94beb46f4        2 months ago        417MB
amazon/aws-sam-cli-emulation-image-nodejs10.x                                                       rapid-1.0.0                                              57fb8d5be66c        2 months ago        361MB
amazon/aws-sam-cli-emulation-image-nodejs12.x                                                       rapid-1.0.0                                              b64e2bfa4562        2 months ago        366MB
amazon/aws-sam-cli-emulation-image-python3.7                                                        rapid-1.2.0                                              95a05d06fb87        5 days ago          889MB
amazon/aws-sam-cli-emulation-image-python3.6                                                        rapid-1.2.0                                              9531da666779        5 days ago          849MB
```

**Comment by awood45** [comment]
This came from a conversation I had with Austin. An additional facet to this is that these image tags sometimes mask updates to the underlying emulation image. A feature request could also be for us to occasionally detect and help force updates when `--skip-pull-image` is not set.

**Comment by johnnatan-jarac** [comment]
Hello, 

I am getting the error: **"amcli.local.docker.manager.DockerImagePullFailedException: Could not find amazon/aws-sam-cli-emulation-image-java8:rapid-1.15.0 image locally and failed to pull it from docker"**, In fact the tag **"rapid-1.15.0"** doesn't exist in Docker Hub. How can I fix that?

Thanks in advance!!

**Comment by mildaniel** [comment]
Closing as this was resolved by #3092.
